src/core/Akka/IO/Buffers/DirectBufferPool.cs

```diff
@@ -195,8 +224,10 @@ public IEnumerable<ByteBuffer> Rent(int minimumSize)
         public void Release(ByteBuffer buf)
         {
             // only release buffers that have actually been taken from one of the segments
-            if (buf.Count == _bufferSize && _segments.Contains(buf.Array))
-                _buffers.Push(buf);
+            if (buf.Count != _bufferSize || !_segments.Contains(buf.Array))
+                throw new Exception("Wrong ArraySegment<byte> was released to DirectBufferPool");
```

Comment:
Sanity check to make sure that any `ArraySegment<byte>` returned to the buffer actually belongs to the buffer, should only happen during development.

Reply:
Why would it only happen during development?